src/main/java/de/unistuttgart/bugfinder/gameresult/GameResultService.java

```diff
@@ -0,0 +1,200 @@
+package de.unistuttgart.bugfinder.gameresult;
+
+import de.unistuttgart.bugfinder.code.Code;
+import de.unistuttgart.bugfinder.configuration.Configuration;
+import de.unistuttgart.bugfinder.configuration.ConfigurationRepository;
+import de.unistuttgart.bugfinder.solution.Solution;
+import de.unistuttgart.bugfinder.solution.SolutionDTO;
+import de.unistuttgart.bugfinder.solution.SolutionRepository;
+import de.unistuttgart.bugfinder.solution.bug.Bug;
+import de.unistuttgart.bugfinder.solution.bug.BugDTO;
+import feign.FeignException;
+import java.util.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+@Slf4j
+public class GameResultService {
+
+  static final long MAX_SCORE = 100l;
+
+  @Autowired
+  private ConfigurationRepository configurationRepository;
+
+  @Autowired
+  private SolutionRepository solutionRepository;
+
+  @Autowired
+  private ResultClient resultClient;
+
+  public void saveGameResult(final GameResultDTO gameResultDTO, final String userId) {
+    final OverworldResultDTO resultDTO = new OverworldResultDTO(
+      "BUGFINDER",
+      gameResultDTO.getConfigurationId(),
+      calculateScore(gameResultDTO),
+      userId
+    );
+    try {
+      resultClient.submit(resultDTO);
+    } catch (final FeignException.BadGateway badGateway) {
+      final String warning =
+        "The Overworld backend is currently not available. The result was NOT saved. Please try again later";
+      log.warn(warning + badGateway);
+      throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, warning);
+    } catch (final FeignException.NotFound notFound) {
+      final String warning = "The result could not be saved. Unknown User";
+      log.warn(warning + notFound);
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, warning);
+    }
+  }
+
+  /**
+   * Calculates the score of a game result which contains evaluating the solution of every code to calculate the score.
+   *
+   * @param gameResultDTO the players game result
+   * @return a score between 0 and 100
+   */
+  private long calculateScore(final GameResultDTO gameResultDTO) {
+    final Configuration configuration = configurationRepository
+      .findById(gameResultDTO.getConfigurationId())
+      .orElseThrow(() ->
+        new ResponseStatusException(
+          HttpStatus.NOT_FOUND,
+          String.format("Configuration with id %s not found.", gameResultDTO.getConfigurationId())
+        )
```

Comment:
Can we please offload this block into a method, that only takes one parameter, the ID to load?
I've seen this call **way** to often.